app/state/[state]/session/[session]/page.tsx

```diff
@@ -1,13 +1,20 @@
 import { NextResponse } from 'next/server';
 import transformData from "@/utilities/transformMasterList";
 import Bill from '@/data/bills';
+import PaginationControls from './paginationControls';
+
+const getPage = (pageSize: number, page: number, list: any[]) => {
+    const start = (page - 1) * pageSize;
+    const end = start + pageSize;
+    return list.slice(start, end);
+};
 
 async function getSessionBills(sessionID: number): Promise<Bill[]> {
     try {
-        const legiscanApiKey = process.env.LEGI_KEY; 
+        const legiscanApiKey = process.env.LEGI_KEY;
         const res = await fetch(`https://api.legiscan.com/?key=${legiscanApiKey}&op=getMasterList&id=${sessionID}`);
```

Comment:
`getSearch` does support paging so I think we should implement that here instead of using `getMasterList`. To get around the lack of session data from `getSearch`, we can implement a second async function to fetch session data by utilizing the `getSessionPeople` legibill operation that returns both the session object and "people" involved in the session (we can ignore the people array for now but might be cool to have a /session/[session]/people page in the future!). 

for ex session/[session]/page.tsx will has two separate functions to handle bills and session data: 
```typescript
export default async function Page({ params, searchParams }: { params: { session: number; state: string }, searchParams: { page: string } }) {
    // getSessionBills uses legibills getSearch endpoint and leverages their paging support
    const billList: someType = await getSessionBills(params.session, otherParamsLikePageAndPageSize)
    // getSessionData uses getSessionPeople which will give you back an object that contains the session data
    const sessionData: someType = await getSessionData(params.session)

    // other stuffz
}
```

It would be good to model the types correctly here. After you implement these changes, I'll come in and clean up the types before merging.

Reply:
Also, you may need to call `getSessionsBill` from a child component or create a layout for this page and call `getSessionData` from that to prevent fetching session data every time a page changes @JonathanTylerWeber